src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -245,7 +249,7 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_SetThreadProvis
     ChipDeviceEvent event;
     event.Type                                           = DeviceEventType::kServiceProvisioningChange;
     event.ServiceProvisioningChange.IsServiceProvisioned = true;
-    PlatformMgr().PostEvent(&event);
+    ReturnErrorOnFailure(PlatformMgr().PostEvent(&event));
 
     return MapOpenThreadError(otErr);
```

Comment:
If otErr is a failure _and_ we fail to post the event (which seems like an odd thing to be doing if otErr is a failure!), which error should we return?

My gut feeling is that we should only try to post the event if otErr is success.